File: SBSAnimoji/Pages/Main/PuppetThumbnailCollectionViewCell.m
//
//  PuppetThumbnailCollectionViewCell.m
//  SBSAnimoji
//
//  Created by Simon Støvring on 05/11/2017.
//  Copyright © 2017 SimonBS. All rights reserved.
//

#import "PuppetThumbnailCollectionViewCell.h"

@interface PuppetThumbnailCollectionViewCell ()
@property (nonatomic, strong) UIImageView *thumbnailImageView;
@property (nonatomic, strong) UIImageView *selectionImageView;
@end

@implementation PuppetThumbnailCollectionViewCell

- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:CGRectZero]) {
        [self setupView];
    }
    return self;
}

- (void)setupView {
    self.backgroundColor = [UIColor whiteColor];
    
    self.thumbnailImageView = [[UIImageView alloc] init];
    self.thumbnailImageView.translatesAutoresizingMaskIntoConstraints = NO;
    self.thumbnailImageView.contentMode = UIViewContentModeScaleAspectFit;
    [self.contentView addSubview:self.thumbnailImageView];
    
    self.selectionImageView = [[UIImageView alloc] init];
    self.selectionImageView.image = [UIImage imageNamed:@"selection"];
    self.selectionImageView.translatesAutoresizingMaskIntoConstraints = NO;
    self.selectionImageView.hidden = YES;
    [self.contentView addSubview:self.selectionImageView];
}

- (void)layoutSubviews {
    [super layoutSubviews];
    UIEdgeInsets thumbnailMargins = UIEdgeInsetsMake(7, 7, 7, 7);
    self.thumbnailImageView.frame = UIEdgeInsetsInsetRect(self.contentView.bounds, thumbnailMargins);
    self.selectionImageView.frame = self.contentView.bounds;
}

- (void)setSelected:(BOOL)selected {
    [super setSelected:selected];
    self.selectionImageView.hidden = !selected;
}

@end

File: Animoji/Model/PuppetThumbnailCollectionViewCell.swift
//
//  PuppetThumbnailCollectionViewCell.swift
//  Animoji
//
//  Created by Daniel Illescas Romero on 25/12/2017.
//  Copyright © 2017 Daniel Illescas Romero. All rights reserved.
//

import UIKit

@objc class PuppetThumbnailCollectionViewCell: UICollectionViewCell, UICollectionViewDelegate {
	
	var thumbnailImageView = UIImageView()
	var selectionImageView = UIImageView()
	
	override init(frame: CGRect) {
		super.init(frame: CGRect.zero)
		self.setupView()
	}
	
	required init?(coder aDecoder: NSCoder) {
		fatalError("init(coder:) has not been implemented")
	}
	
	func setupView() {
		
		self.backgroundColor = .white
		
		self.thumbnailImageView.translatesAutoresizingMaskIntoConstraints = false
		self.thumbnailImageView.contentMode = .scaleAspectFit
		self.contentView.addSubview(self.thumbnailImageView)
		
		self.selectionImageView.image = #imageLiteral(resourceName: "selection")
		self.selectionImageView.translatesAutoresizingMaskIntoConstraints = false
		self.selectionImageView.isHidden = true
		self.contentView.addSubview(self.selectionImageView)
	}
	
	override func layoutSubviews() {
		super.layoutSubviews()
		let thumbnailMargins = UIEdgeInsetsMake(7, 7, 7, 7)
		self.thumbnailImageView.frame = UIEdgeInsetsInsetRect(self.contentView.bounds, thumbnailMargins)
		self.selectionImageView.frame = self.contentView.bounds
	}
	
	override var isSelected: Bool {
		didSet {
			self.selectionImageView.isHidden = !isSelected
		}
	}
}
